pkg/foreach/R/onLoad.R

```diff
@@ -0,0 +1,21 @@
+#
+# Copyright (c) Microsoft. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+.onLoad <- function(libname, pkgname) {
+  local <- as.logical(Sys.getenv("R_FOREACH_DOPAR_LOCAL", "FALSE"))
```

Comment:
Use also:
```r
local <- as.logical(Sys.getenv("R_FOREACH_DOPAR_LOCAL", "FALSE"))
local <- getOption("foreachDoparLocal", local)
options(foreachDoparLocal=local)
```
So you don't override what user might already have set via `options()`.